### DISCUSSION TITLE
Windows container support

### DISCUSSION THREAD

**Discussion by patrick-stephens** [fluent/fluent-bit#5007] [General]
Is anyone using the Windows container definitions from Fluent Bit?

As part of the Windows CI migration I've started bringing the Windows builds into Github Actions. 
https://github.com/fluent/fluent-bit/blob/a703837eefc37258a2004a937b64dd08df26d04f/.github/workflows/windows-build.yaml

The Windows containers are built on push to `master` now but they are very large (almost 6Gb) and very slow to build (>1hr).
See https://github.com/fluent/fluent-bit/actions/runs/1946115093 as an example.

If we remove the building of Windows containers then we do not test they actually build and so keeping the definitions in the OSS repo seems wrong: it'll just be a snapshot of them last time they did build and may be wildly different now. 

Should we remove the Windows container definitions or keep building them or move to another bit of job, and can they be optimised?

If we do build them, what specific targets do people need?

**Comment by lubingfeng** [comment]
Interested in hearing the need

**Comment by ilanitr** [comment]
`Anthos` is using the Windows definitions from Fluent Bit. We have a pipeline to sync from OSS repo and on each change (or branch cut), we are creating an `Anthos` specific tag and build an image, based on that tag, using this dockerfile definitions (pushed to `gcr` and used in our bundle). 
We are experiencing  failures to build 1.8.14 and 1.8.15 due to the missing dockerfile. For future patches, I would like to add the `Dockerfile.windows` file back to the `dockerfiles` folder. I'll raise a PR for that.

**Reply by patrick-stephens** [reply]
OK, so we'll keep them in and build them - there's no point having definitions we don't build as who knows if they work then.

What targets are you using/need?

**Reply by ilanitr** [reply]
For 1.8 we are using `dockerfiles/Dockerfile.windows.devel`

I'll send a PR to add it back to the branch :)

Thanks!

**Comment by patrick-stephens** [comment]
For anyone watching, Windows containers are now pushed to Docker Hub for releases.
